### PR TITLE
Fix being able to save on no changes in drawer-item

### LINKED ISSUE #19417: Data model editing: Follow o2m/m2m alias field to another model enters modified state
### Describe the Bug

Click the link of an o2m/m2m alias field
image
It goes to the related model, however the model editing page is modified (but nothing is modified) 
image
Goes back would trigger the confirm dialog:
image

This happens for both o2m and m2m alias field

### To Reproduce

as above.

### Directus Version

v10.5.3

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by Nitwel**
~~attempted #19417~~

You were able to press save in the drawer item even though there where no changes to be saved.

An alternative fix would have been to validate the changes on the v-form side but that is overly complex especially because of the `{create, update, delete}` structure of relations.

### CODE REVIEW

**Review by paescuj — CHANGES_REQUESTED** [review]
Could we apply the same logic to `drawer-collection`?

File: app/src/views/private/components/drawer-item.vue [hunk 1]
```diff
@@ -151,7 +151,15 @@ const swapFormOrder = computed(() => {
 	return props.junctionFieldLocation === 'top';
 });
 
-const hasEdits = computed(() => !isEmpty(internalEdits.value));
+const hasEdits = computed(() => {
+	if (isEmpty(internalEdits.value)) return false;
+
+	if (props.junctionField && Object.keys.length === 1 && isEmpty(internalEdits.value[props.junctionField]))
```
Comment (paescuj) [inline app/src/views/private/components/drawer-item.vue:157]: What does `Object.keys.length === 1` do?

Comment (Nitwel) [reply app/src/views/private/components/drawer-item.vue:157]: Always equal to true. 😅

**Comment by Nitwel** [comment]
Not directly as the drawer-collection is only for selecting items. But what I did is set value.value = []; instead of undefined which was causing the save button to be available even if there where no items selected.

**Comment by Nitwel** [comment]
The most appropiate fix for drawer-collection now was to check on the form-field if it is a relational field and the value gets set to undefined.

**Review by br41nslug — CHANGES_REQUESTED** [review]
Im not sure this PR fixes the linked issue. The original issue explicitly mentions following the link in the data-model to another collection (which does not involve the items-drawer) which still shows the blocking save behavior described there 

https://github.com/directus/directus/assets/9389634/04f6042e-97cb-4d66-a716-3b1990b88f46

File: app/src/components/v-form/form-field.vue [hunk 2]
```diff
@@ -154,7 +154,12 @@ const validationPrefix = computed(() => {
 
 function emitValue(value: any) {
 	if (
-		(isEqual(value, props.initialValue) || (props.initialValue === undefined && isEqual(value, defaultValue.value))) &&
+		(isEqual(value, props.initialValue) ||
+			(props.initialValue === undefined && isEqual(value, defaultValue.value)) ||
+			(props.field.meta?.special?.some((s) =>
+				['m2m', 'm2o', 'o2m', 'translations', 'm2a', 'file', 'files'].includes(s)
```
[the remark this answers is not in the document]
Comment (rijkvanzanten) [reply app/src/components/v-form/form-field.vue:160]: Yes.

**Comment by br41nslug** [comment]
Im closing this PR since it doesn't fix the originally linked issue and I am not directly sure which issue it does fix.

**Pull request closed without merging**